components/sync_manager/src/manager.rs

```diff
@@ -238,22 +238,31 @@ impl SyncManager {
         let key_bundle = sync15::KeyBundle::from_ksync_base64(&params.acct_sync_key)?;
         let tokenserver_url = url::Url::parse(&params.acct_tokenserver_url)?;
 
-        let logins_sync = should_sync(&params, LOGINS_ENGINE);
-        let bookmarks_sync = should_sync(&params, BOOKMARKS_ENGINE);
-        let history_sync = should_sync(&params, HISTORY_ENGINE);
-        let tabs_sync = should_sync(&params, TABS_ENGINE);
+        let bookmarks_sync = should_sync(&params, BOOKMARKS_ENGINE) && places.is_some();
+        let history_sync = should_sync(&params, HISTORY_ENGINE) && places.is_some();
+        let logins_sync = should_sync(&params, LOGINS_ENGINE) && logins.is_some();
+        let tabs_sync = should_sync(&params, TABS_ENGINE) && tabs.is_some();
 
         let places_conn = if bookmarks_sync || history_sync {
             places
                 .as_mut()
-                .expect("already checked")
+                .expect("trying to sync an engine that has not been configured")
                 .open_sync_connection()
                 .ok()
         } else {
             None
         };
-        let l = logins.as_ref().map(|l| l.lock().expect("poisoned mutex"));
-        let t = tabs.as_ref().map(|t| t.lock().expect("poisoned mutex"));
+        let l = if logins_sync {
+            logins.as_ref().map(|l| l.lock().expect("poisoned mutex"))
+        } else {
+            None
+        };
```

Comment:
Further down in this file, we were doing `assert!(logins_sync, "Should have already checked");` but it wasn't clear to me what had already been checked. Nothing was actually checking `logins_sync` except for that assertion, which makes me wonder if we would always attempt to sync logins and tabs even if not explicitly requested by the app.